uservice_ccutter/server.py

```diff
@@ -104,88 +104,103 @@ def post_request(ptype):
                 userdict[fld] = inputdict[fld]
             # Here's the magic.
             substitute(ptype, auth, userdict)
-            retval = create_project(ptype, auth, userdict)
-            finalize(ptype, auth, userdict)
+            # finalize_ may need to do work with checked-out repo
+            with TempDir() as workdir:
+                retval = create_project(ptype, auth, userdict, workdir)
+                # This is the point of no return.  We have a GitHub repo,
+                #  which we must report to the user.
+                # Therefore, if finalize raises an exception (it shouldn't)
+                #  we must catch it and wrap it.
+                pce = None
+                # pylint: disable=broad-except
+                try:
+                    pce = finalize(ptype, auth, userdict)
+                except Exception as exc:
+                    # Just report error type and message.
+                    pce = exc.__class__.__name__ + ": " + str(exc)
+                retval["post_commit_error"] = pce
             return jsonify(retval)
 
         # pylint: disable=too-many-locals
-        def create_project(ptype, auth, userdict):
+        def create_project(ptype, auth, userdict, workdir):
```

Comment:
I tried a couple times, but the problem is that there's an intricate dance going on between the local-git and the GitHub-git pieces, and either I'm passing tons of state back and forth, or I just do it in the same (long) function.